src/package/nuspec/Microsoft.NET.Test.Sdk.targets

```diff
@@ -76,15 +76,28 @@
       <Line Include="End Sub"/>
       <Line Include="End Module"/>
     </ItemGroup>
+
+    <ItemGroup Condition="'$(Language)'=='F#'">
+      <Line Include="// &lt;auto-generated&gt; This file has been auto generated. &lt;/auto-generated&gt;"/>
+      <Line Include="module AutoGeneratedProgram"/>
+      <Line Include="[&lt;EntryPoint&gt;]"/>
+      <Line Include="[&lt;Microsoft.VisualStudio.TestPlatform.TestSDKAutoGeneratedCode&gt;]"/>
+      <Line Include="let main _ = 0"/>
+    </ItemGroup>
+
     <WriteLinesToFile
       File="$(GeneratedProgramFile)" Lines="@(Line)" Overwrite="true" Encoding="Unicode">
     </WriteLinesToFile>
 
     <!--
     Compile Include the generated Program File
     -->
-    <ItemGroup>
+    <ItemGroup Condition="'$(Language)'!='F#'">
       <Compile Include="$(GeneratedProgramFile)"/>
     </ItemGroup>
+
+    <ItemGroup Condition="'$(Language)'=='F#'">
```

Comment:
I'm not sure `runtimeconfig.*.json` will gets generates because `CompileAfter`?
FYI https://github.com/dotnet/sdk/issues/201 https://github.com/Microsoft/vstest/issues/287.

@saul did you test simple unit test project of F# with your changes?

Reply:
I think I understood what is [CompileAfter](https://github.com/Microsoft/visualfsharp/blob/61907dcd859702924021bc0aad74f64ce459c9f0/src/fsharp/FSharp.Build/Microsoft.FSharp.Targets#L296). Now it makes sense after reading issue one more time. 

I have checked out your branch and verified E2E scenario. Below are the issue we need to address.
-  Newly created F# unit test project not getting compiled because of below compilation error.
```ps
PS C:\Users\samadala\tmp\fs-test-project> dotnet test
Build started, please wait...
C:\Users\samadala\tmp\fs-test-project\Program.fs(1,1): error FS0222: Files in libraries or multiple-file applications must begin with a namespace or module declaration. When using a module declaration at the start of a file the '=' sign is not allowed. If this is a top-level module, consider removing the = to resolve this error. [C:\Users\samadala\tmp\fs-test-project\fs-test-project.fsproj]
```
This can be fixed by removing `<Compile Include="Program.fs" />` from template fsproj [here](https://github.com/dotnet/templating/blob/4cc28db23041bfbeae2797fbfbc7356b50c09e84/template_feed/Microsoft.DotNet.Test.ProjectTemplates.2.0/content/MSTest-FSharp/Company.TestProject1.fsproj#L14) and removing [Program.fs](https://github.com/dotnet/templating/blob/4cc28db23041bfbeae2797fbfbc7356b50c09e84/template_feed/Microsoft.DotNet.Test.ProjectTemplates.2.0/content/MSTest-FSharp/Program.fs), We can address this with separate https://github.com/dotnet/templating PR 

- Test project created with old templates fail to compile with above error on updating to latest `Microsoft.NET.Test.Sdk` package.

This can be fixed by adding check [here](https://github.com/Microsoft/vstest/pull/1664/files#diff-251b10661dfb4f6d4d69fb78fe9127b6R99) that don't include  `GeneratedProgramFile` in `CompileAfter` if `Compile` item group contains `Program.fs`.

@saul Can you please fix the seconds issue in this PR?

Reply:
Thanks @smadala - I've fixed the second problem.

I've added a warning: `A 'Program.fs' file can be automatically generated for F# .NET Core test projects. To fix this warning, either delete the file from the project, or set the <GenerateProgramFile> property to 'false'.`

This lets the user either remove their Program.fs, or disables `<GenerateProgramFile>` if they need a complex Program.fs for whatever reason.

Reply:
@saul Adding warning is very good idea.